### PR TITLE
releng/handbook: Add documentation for image promotion during the release process

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this:
/kind documentation

#### What this PR does / why we need it:
Add documentation regarding the Image Promotion process in the branch manager handbook

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/cc @kubernetes/release-managers 
/assign @justaugustus @saschagrunert

### CODE REVIEW

File: release-engineering/role-handbooks/branch-manager.md [hunk 1]
```diff
@@ -283,6 +284,8 @@ Builds against the `master` branch are implicitly the next alpha. `krel gcbmgr`
 
 ### Beta Releases
 
+Before run the `nomock` please refeer to the [Image Promotion documentation](#image-promotion).
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:287]:
```suggestion
Before run the `nomock stage` please refer to the [Image Promotion documentation](#image-promotion).
```

File: release-engineering/role-handbooks/branch-manager.md [hunk 2]
```diff
@@ -293,6 +296,8 @@ Builds against a `release-x.y` branch are implicitly the next RC (release candid
 
 **Note: If this is the first release (`rc.0`), there are additional tasks to complete. Please review them _COMPLETELY_ in the [Branch Creation section](#branch-creation), _before_ continuing.**
 
+Before run the `nomock` please refeer to the [Image Promotion documentation](#image-promotion).
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:299]:
```suggestion
Before run the `nomock stage` please refer to the [Image Promotion documentation](#image-promotion).
```

File: release-engineering/role-handbooks/branch-manager.md [hunk 3]
```diff
@@ -330,6 +335,8 @@ Otherwise we might have a mix of PRs against master, some have been merged in co
 
 ### Official Releases
 
+Before run the `nomock` please refeer to the [Image Promotion documentation](#image-promotion).
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:338]:
```suggestion
Before run the `nomock stage` please refer to the [Image Promotion documentation](#image-promotion).
```

File: release-engineering/role-handbooks/branch-manager.md [hunk 4]
```diff
@@ -373,6 +380,39 @@ The following are some ways to determine if the release process was successful:
 
 3. CHANGELOG-X.Y.md is automatically loaded into the kubernetes/kubernetes repo: [https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md)
 
+### Image Promotion
+
+ We need publishing images to the `k8s-staging-kubernetes` and then promoting them to prodution. To do that we need to follow the steps:
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:385]:
```suggestion
 We need publishing images to the `k8s-staging-kubernetes` GCS Bucket and then promoting them to production. To do that we need to follow the steps:
```

File: release-engineering/role-handbooks/branch-manager.md [hunk 5]
```diff
@@ -373,6 +380,39 @@ The following are some ways to determine if the release process was successful:
...
+
+ This step should be done after `Mock Stage` and `Mock Release` are completed and no errors.
+
+ - Forking/cloning https://github.com/kubernetes-sigs/k8s-container-image-promoter
+ - Forking/cloning https://github.com/kubernetes/k8s.io. If you already have that make sure you have an up-to-date copy of the repository.
+ - From the root of `k8s-container-image-promoter` repository
+
+ ```shell
+ go install ./cmd/cip-mm/... # Installs the manifest merge utility, cip-mm
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:394]: Can we `get get …` the promoter? If so, then I would prefer this way.

File: release-engineering/role-handbooks/branch-manager.md [hunk 6]
```diff
@@ -373,6 +380,39 @@ The following are some ways to determine if the release process was successful:
...
+_note 2_: The `--filter_tag` you will add the release that you are working, like `v1.19.0-rc.3` / `v1.19.0-beta.1` / `v1.19.0`
+
+ - After the command ran you should see updates to the `k8s.gcr.io/images/k8s-staging-kubernetes/images.yaml`
+ - Create a branch and open a Pull Request with the new digests to promote. Add the links for cloudbuild jobs for `Mock Stage` and `Mock Release` and tag `@kubernetes/release-engineering` and add a explict `/hold` to be lifted before we start the `Official Release` step.
+
+ Examples:
+ - [PR example 1](https://github.com/kubernetes/k8s.io/pull/1049)
+ - [PR example 2](https://github.com/kubernetes/k8s.io/pull/1071)
+
+ - After the Pull Request is approved and **before** start the `Official Release` step, we need to unhold the PR to get that merged  and watch the following [Prow Job](https://prow.k8s.io/?job=post-k8sio-cip) when the latest master ran without errors we can kick off the `Official Release`.
```
Comment (saschagrunert) [inline release-engineering/role-handbooks/branch-manager.md:413]:
```suggestion
 - After the Pull Request is approved and **before** start the `Official Release` step, we need to unhold the PR to get that merged  and watch the following [Prow Job](https://prow.k8s.io/?job=post-k8sio-cip) to succeed. When the latest master ran without errors, then we can continue with `Official Release`.
```

**Comment by tpepper** [comment]
Need to re-evaluate versus https://github.com/kubernetes/release/pull/1455

**Comment by justaugustus** [comment]
/hold until https://github.com/kubernetes/release/pull/1457 is merged and I can give a review on updated process.
/retitle releng/handbook: Add documentation for image promotion during the release process

File: release-engineering/role-handbooks/branch-manager.md (hunk 5 above)
Comment (cpanato) [reply release-engineering/role-handbooks/branch-manager.md]: tried but got
```
GO111MODULE=on go get sigs.k8s.io/k8s-container-image-promoter/cmd/cip-mm
go get sigs.k8s.io/k8s-container-image-promoter/cmd/cip-mm: module sigs.k8s.io/k8s-container-image-promoter@upgrade found (v1.0.7), but does not contain package sigs.k8s.io/k8s-container-image-promoter/cmd/cip-mm
```

**Comment by cpanato** [comment]
@justaugustus done.

**Comment by justaugustus** [comment]
/lgtm
/approve
/hold cancel

I have some grammar nits, but let's address that in a follow-up.

**Pull request merged**